agent/agent.go

```diff
@@ -3992,6 +3996,9 @@ func (a *Agent) loadLimits(conf *config.RuntimeConfig) {
 	a.config.RPCMaxBurst = conf.RPCMaxBurst
 }
 
+// ReloadConfig will atomically reload all configs from the given newCfg,
+// including all services, checks, tokens, metadata, dnsServer configs, etc.
+// It will also reload all ongoing watches.
 func (a *Agent) ReloadConfig(newCfg *config.RuntimeConfig) error {
```

Comment:
This method seemed like a doozy. I'm sure there's some context that should be provided here (perhaps historical? or design related?). Let me know what it is and I'm happy to update the PR!